C/Cling/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# Cling is an interactive C++ interpreter,
+# built on the top of LLVM and Clang libraries.
+# https://root.cern/cling
+
+name = "cling"
+version = v"0.9.0"
+
+# Collection of sources required to complete build
+# Cling requires CERN's patched versions of llvm and clang.
+sources = [
+    GitSource("http://root.cern/git/llvm.git", "859a19cc70aba64df3a20f1b91c84b81547bbf24"),
+    GitSource("http://root.cern/git/clang.git", "0e0d27eb409f64af84cc6a85c2fb48d2b5ce8122"),
+    GitSource("http://root.cern/git/cling.git", "646e3f59550b6a24d81edde5694844a1a8c3fdf4")
+]
+
+# Bash recipe for building across all platforms
+# Cling Build Instructions: https://root.cern/cling/cling_build_instructions/
+# Cling Homebrew Formula: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/cling.rb
+script = raw"""
+cd $WORKSPACE/srcdir
+mv llvm/ src
+mv clang/ src/tools/
+mv cling/ src/tools/
+mkdir build
+cd build/
+cmake -DCMAKE_INSTALL_PREFIX=$prefix -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release LLVM_BUILD_TOOLS=OFF ../src
```

Comment:
Did you mean
```suggestion
cmake -DCMAKE_INSTALL_PREFIX=$prefix \
    -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
    -DCMAKE_BUILD_TYPE=Release \
    -DLLVM_BUILD_TOOLS=OFF \
    ../src
```
(note the `-D` before `LLVM_BUILD_TOOLS`)?

Reply:
Yes.
(It builds for me anyway under Docker on intel macOS)

I've already changed things a bit on my local branch because of this: https://github.com/JuliaPackaging/Yggdrasil/blob/86a4221f71ee8ca5c0fd1056829a1fe2b28c879b/M/Metal_LLVM_Tools/build_tarballs.jl#L35

I currently have something like this (still a work in progress)
```bash
# The very first thing we need to do is to build llvm-tblgen for x86_64-linux-muslc
# This is because LLVM's cross-compile setup is kind of borked, so we just
# build the tools natively ourselves, directly.  :/
# See: https://github.com/JuliaPackaging/Yggdrasil/blob/86a4221f71ee8ca5c0fd1056829a1fe2b28c879b/M/Metal_LLVM_Tools/build_tarballs.jl#L35

# Build llvm-tblgen and llvm-config
mkdir ${WORKSPACE}/bootstrap
pushd ${WORKSPACE}/bootstrap
CMAKE_FLAGS=()
CMAKE_FLAGS+=(-DLLVM_TARGETS_TO_BUILD:STRING=host)
CMAKE_FLAGS+=(-DLLVM_HOST_TRIPLE=${MACHTYPE})
CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)
CMAKE_FLAGS+=(-DLLVM_ENABLE_PROJECTS='llvm')
CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=False)
CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_HOST_TOOLCHAIN})
cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
ninja -j${nproc} llvm-tblgen llvm-config
popd

# Let's do the actual build within the `build` subdirectory
mkdir ${WORKSPACE}/build && cd ${WORKSPACE}/build
CMAKE_FLAGS=()

# Tell LLVM where our pre-built tblgen tools are
CMAKE_FLAGS+=(-DLLVM_TABLEGEN=${WORKSPACE}/bootstrap/bin/llvm-tblgen)
CMAKE_FLAGS+=(-DLLVM_CONFIG_PATH=${WORKSPACE}/bootstrap/bin/llvm-config)

# Install things into $prefix
CMAKE_FLAGS+=(-DCMAKE_INSTALL_PREFIX=${prefix})

# Explicitly use our cmake toolchain file and tell CMake we're cross-compiling
CMAKE_FLAGS+=(-DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN})
CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING:BOOL=ON)

# Release build for best performance
CMAKE_FLAGS+=(-DCMAKE_BUILD_TYPE=Release)

CMAKE_FLAGS+=(-DLLVM_TARGETS_TO_BUILD="host;NVPTX")

CMAKE_FLAGS+=(-DLLVM_BUILD_TOOLS=OFF)

cmake -GNinja ${LLVM_SRCDIR} ${CMAKE_FLAGS[@]}
ninja -j${nproc} \
    tools/cling/install \
    install-clang-resource-headers
```

Reply:
```
CMAKE_FLAGS+=(-DCMAKE_CROSSCOMPILING=False)
```
looks like a blatant lie.

Reply:
I'm copying this stuff from ...

here: https://github.com/JuliaPackaging/Yggdrasil/blob/73a24b84398c08df17b8f90028f5257a6668cdd8/L/LLVM/common.jl#L72-L74

and here:
https://github.com/JuliaPackaging/Yggdrasil/blob/86a4221f71ee8ca5c0fd1056829a1fe2b28c879b/M/Metal_LLVM_Tools/build_tarballs.jl#L35-L37

It seems like cross compilation is broken for llvm, so these work-arounds are needed.

Reply:
More than LLVM specifically being broken, the problem is that as far as I know in CMake there is no way to pass both host _and_ target toolchains at the same time, like more sensible build systems allow to do, so you have to manually build host tools separately.  I still think manually setting `CMAKE_CROSSCOMPILING` is unnecessary though.